libcontainer/cgroups/fs/memory.go

```diff
@@ -34,10 +38,20 @@ func (s *MemoryGroup) Apply(d *cgroupData) (err error) {
 				return err
 			}
 		}
-		// We have to set kernel memory here, as we can't change it once
-		// processes have been attached to the cgroup.
-		if err := s.SetKernelMemory(path, d.config); err != nil {
-			return err
+
+		// Check if kernel memory is enabled
+		if _, err := os.Stat(filepath.Join(path, cgroupKernelMemoryLimit)); err == nil {
```

Comment:
The idea was we SetKernelMemory() only if kernel memory is enabled. So have the check and then call the method.

But having the check inside the method would be more cleaner. I will make that change.